components/job-orchestration/job_orchestration/executor/query/fs_search_task.py

```
@@ -102,7 +106,15 @@ def _make_core_clp_s_command_and_env_vars(
             archive_id,
         ))
         # fmt: on
-        env_vars = None
+    if not is_telemetry_disabled_by_env():
+        enable_telemetry = random.choices(
+            [True, False],
+            cum_weights=[worker_config.query_worker.search_sampling_probability, 1.0],
+            k=1,
+        )[0]
```

Ruff (0.15.18)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)